### PR TITLE
use jest for tests

### PR DESCRIPTION
**Pull request by lennyburdette**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -17,7 +17,11 @@
         "${workspaceFolder}/**",
         "!**/node_modules/**"
       ],
-      "args": [""]
+      "args": ["graphene"],
```
Comment (lennyburdette) [inline .vscode/launch.json:20]: did you mean to narrow this to just graphene?

File: src/tests/provides.test.ts [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
       query ($id: ID!) {
         product(id: $id) {
           createdBy { email totalProductsCreated }
         }
       }`,
-    variables: { id: "apollo-federation" },
+    variables: { id: 'apollo-federation' },
   });
 
+  const totalProductsCreated: number =
```
Comment (lennyburdette) [inline src/tests/provides.test.ts:14]: move this line after the first expectation ... if the response doesn't contain `data` it'll blow up with a confusing "property product does not exist on undefined" error

Comment (lennyburdette) [reply src/tests/provides.test.ts:14]: (or use `?.` everywhere)

File: src/tests/provides.test.ts [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
+    resp.data.product.createdBy.totalProductsCreated;
+
   expect(resp).toMatchObject({
     data: {
       product: {
         createdBy: {
-          email: "support@apollographql.com",
-          totalProductsCreated: 1337,
+          email: 'support@apollographql.com',
+          totalProductsCreated: expect.any(totalProductsCreated),
```
Comment (lennyburdette) [inline src/tests/provides.test.ts:22]: `expect.any(Number)`

File: .vscode/launch.json (hunk 1 above)
Comment (michael-watson) [reply .vscode/launch.json:20]: good catch

File: src/tests/provides.test.ts (hunk 2 above)
Comment (michael-watson) [reply src/tests/provides.test.ts:14]: 👍

File: src/tests/provides.test.ts (hunk 3 above)
Comment (michael-watson) [reply src/tests/provides.test.ts:22]: 👍

**Pull request merged**